cmd/collect-profiles/main.go

```diff
@@ -0,0 +1,364 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/operator-framework-olm/pkg/profiling/config"
+)
+
+const (
+	profileConfigMapLabelKey = "olm.openshift.io/pprof"
+)
+
+var (
+	rootCmd = newCmd()
+
+	// Used for flags
+	namespace       string
+	configMountPath string
+	certMountPath   string
+)
+
+func init() {
+	rootCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", "default", "The Kubernetes namespace where the generated configMaps should exist. Defaults to \"default\".")
+	rootCmd.MarkFlagRequired("namespace")
+	rootCmd.PersistentFlags().StringVarP(&configMountPath, "config-mount-path", "c", "/etc/config", "The path to the collect-profiles configuration file.")
+	rootCmd.MarkFlagRequired("config-mount-path")
+	rootCmd.PersistentFlags().StringVarP(&certMountPath, "cert-mount-path", "", "/var/run/secrets/serving-cert", "The path to the tls cert used by the client making https requests against the pprof URLs.")
+}
+
+func main() {
+	Execute()
+}
+
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		klog.Fatal(err)
+		os.Exit(1)
+	}
+}
+
+func getTruePointer() *bool {
+	trueBool := true
+	return &trueBool
+}
+
+func newCmd() *cobra.Command {
+	var cfg config.Configuration
+	return &cobra.Command{
+		Use:   "collect-profiles configMapName:url",
```

Comment:
It looks like we already have a relatively fleshed out set of flags. Any reason the configmapName and URL arguments cannot also be flags, or is this kind of setup simply preference?

Reply:
Ah, I realize this is because we can provide a series of arguments. Might be still worth fleshing out the long description then at some point.

Reply:
>Ah, I realize this is because we can provide a series of arguments. Might be still worth fleshing out the long description then at some point.

Agreed.

Reply:
Addressed in the latest commit.